app/models/lettings_log.rb

```diff
@@ -53,6 +53,15 @@ class LettingsLog < Log
   scope :filter_by_before_startdate, ->(date) { where("lettings_logs.startdate >= ?", date) }
   scope :unresolved, -> { where(unresolved: true) }
   scope :filter_by_organisation, ->(org, _user = nil) { where(owning_organisation: org).or(where(managing_organisation: org)) }
+  scope :duplicate_logs, lambda { |log|
+                           attrs_to_check = %w[tenancycode propcode startdate age1 sex1 ecstat1 tcharge postcode_full]
+                           where(log.slice(*attrs_to_check))
+                           .where.not(id: log.id)
+                           .where.not(status: "deleted")
+                           .where.not("startdate IS NULL OR age1 IS NULL OR sex1 IS NULL OR ecstat1 IS NULL OR tcharge IS NULL OR postcode_full IS NULL OR propcode IS NULL OR needstype IS NULL")
+                           .where("location_id = ? OR needstype = 1", log.location_id)
+                         }
+  scope :duplicate_logs_for_organisation, ->(org, log) { filter_by_organisation(org).duplicate_logs(log) }
```

Comment:
I would be happy for you to dodge writing the tests that Nat is asking for by removing this maybe?
I think it would be perfectly readable wherever this is used to just chain the scopes that it combines anyway.

Reply:
If there's a common use case for this I reckon we should keep it in, but if not happy for it to be taken out, up to you :)

Reply:
You're right! We can just call `current_user.lettings_logs.duplicate_logs(@log)` that's way better! Thanks